sdl2promela/sdl/type_info.py

```diff
@@ -0,0 +1,40 @@
+from typing import Dict
+
+from sdl2promela.utils import Asn1Type
+
+
+class TypeInfo:
+
+    all_types: Dict[str, str]
+
+    def __init__(self):
+        self.all_types = {}
+
+    def fix_type_name(self, name: str) -> str:
+        if name in self.all_types:
+            return self.all_types[name]
+        return name
+
+
+# def create_type(name: str, has_parent: bool, t: Any)
+
+
+def build_type_info(all_types: Dict[str, Asn1Type]) -> TypeInfo:
+    type_info = TypeInfo()
+
+    for name, t in all_types.items():
+        if t.AddedType == "False":
```

Comment:
I was confused, because all_types is declared as Dict[str,str].... and then here as Dict[str, Asn1Type]

Reply:
the parameter `all_types` is `Dict[str, Asn1Type]`
The `TypeInfo` class has `all_types` which is `Dict[str, str]`